backend/app/models/question_type.rb

```diff
@@ -3,10 +3,23 @@ class QuestionType < ApplicationRecord
   validates :name, uniqueness: { case_sensitive: false }
   has_many :questions, dependent: :destroy
 
+  before_destroy :check_questions_before_destroy, prepend: true
```

Comment:
I think it will be a good thing for you to read this paper 
https://medium.com/appaloosa-store-engineering/caution-when-using-before-destroy-with-model-association-71600b8bfed2

Reply:
Oh, thanks for pointing that out! Nice Article

Reply:
done!